code/game/objects/items/storage/storage.dm

```diff
@@ -425,6 +425,15 @@
 	if(!can_be_inserted(W, 0 , user))
 		if(contents.len >= storage_slots) //don't use items on the backpack if they don't fit
 			return 1
+
+		var/sum_w_class = W.w_class
+
+		for(var/obj/item/I in contents)

```

Comment:
This should all be in can_be_inserted. Maybe that extra check above is the issue

Reply:
The check is in there but looking at the code I don't know why we don't just return 1 (no afterattack) if an item can't be inserted, rather than all these redundant checks